File: ws/codena/u3/ws29SingletonCode/Javascript/TaxCalculator/tax.js
class Tax {
    static instance = null;

    constructor() {
        if (Tax.instance) {
            return Tax.instance;
        }

        console.log("\nInitializing instance\n");

        this.percentage = 0;
        Tax.instance = this;
    }

    static getInstance() {
        if (Tax.instance === null) {
            Tax.instance = new Tax();
        }

        return Tax.instance;
    }

    updateTaxPercentage(p) {
        this.percentage = p;
    }

    getPercentage() {
        return this.percentage;
    }

    salesTotal(price) {
        return price * this.percentage;
    }
}

export default Tax;

File: ws/codena/u3/ws29SingletonCode/Python/TaxCalculator/tax.py
class Tax:
    __instance = None

    def __new__(cls):
        if cls.__instance is None:
            print("\nInitializing instance\n")
            cls.__instance = super().__new__(cls)
            cls.__instance.percentage = 0
        return cls.__instance

    @staticmethod
    def getInstance():
        return Tax()

    def updateTaxPercentage(self, p):
        self.percentage = p

    def getPercentage(self):
        return self.percentage

    def salesTotal(self, price):
        return price * self.percentage